In [2]:
#import libraries
import pandas as pd
import numpy as np

The dataset we will import is "BankChurners.csv"

In [3]:
# importing dataset
data = pd.read_csv('BankChurners.csv')
# checking the top rows 
data.head()

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0  768805383  Existing Customer            45      M                3   
1  818770008  Existing Customer            49      F                5   
2  713982108  Existing Customer            51      M                3   
3  769911858  Existing Customer            40      F                4   
4  709106358  Existing Customer            40      M                3   

  Education_Level Marital_Status Income_Category Card_Category  \
0     High School        Married     $60K - $80K          Blue   
1        Graduate         Single  Less than $40K          Blue   
2        Graduate        Married    $80K - $120K          Blue   
3     High School        Unknown  Less than $40K          Blue   
4      Uneducated        Married     $60K - $80K          Blue   

   Months_on_book  ...  Months_Inactive_12_mon  Contacts_Count_12_mon  \
0              39  ...                       1                      3   
1              44  ...                       1                      2   
2              36  ...                       1                      0   
3              34  ...                       4                      1   
4              21  ...                       1                      0   

   Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  \
0       12691.0                  777          11914.0                 1.335   
1        8256.0                  864           7392.0                 1.541   
2        3418.0                    0           3418.0                 2.594   
3        3313.0                 2517            796.0                 1.405   
4        4716.0                    0           4716.0                 2.175   

   Total_Trans_Amt  Total_Trans_Ct  Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  
0             1144              42                1.625                  0.061  
1             1291              33                3.714                  0.105  
2             1887              20                2.333                  0.000  
3             1171              20                2.333                  0.760  
4              816              28                2.500                  0.000  

[5 rows x 21 columns]

In [18]:
data.shape

(10127, 21)

Identification of __Datatypes__

In [4]:
data.dtypes

CLIENTNUM                     int64
Attrition_Flag               object
Customer_Age                  int64
Gender                       object
Dependent_count               int64
Education_Level              object
Marital_Status               object
Income_Category              object
Card_Category                object
Months_on_book                int64
Total_Relationship_Count      int64
Months_Inactive_12_mon        int64
Contacts_Count_12_mon         int64
Credit_Limit                float64
Total_Revolving_Bal           int64
Avg_Open_To_Buy             float64
Total_Amt_Chng_Q4_Q1        float64
Total_Trans_Amt               int64
Total_Trans_Ct                int64
Total_Ct_Chng_Q4_Q1         float64
Avg_Utilization_Ratio       float64
dtype: object

## Taking the numerical cols only

Use a variable `numerical_cols` to store the list of `int` and `float` datatypes

In [20]:
# storing indices of all numerical data types in numerical_cols
numerical_cols = data.select_dtypes(exclude=['object']).columns

# checking
numerical_cols

Index(['CLIENTNUM', 'Customer_Age', 'Dependent_count', 'Months_on_book',
       'Total_Relationship_Count', 'Months_Inactive_12_mon',
       'Contacts_Count_12_mon', 'Credit_Limit', 'Total_Revolving_Bal',
       'Avg_Open_To_Buy', 'Total_Amt_Chng_Q4_Q1', 'Total_Trans_Amt',
       'Total_Trans_Ct', 'Total_Ct_Chng_Q4_Q1', 'Avg_Utilization_Ratio'],
      dtype='object')

Find the maximum and minimum of each numerical cols to get a better understanding of the dataset

In [21]:
for col in numerical_cols:
    print("range of {}{}{}{}{}{}{}{}".format(col,":"," ","[",data[col].min(), ", ",data[col].max(),"]"))

range of CLIENTNUM: [708082083, 828343083]
range of Customer_Age: [26, 73]
range of Dependent_count: [0, 5]
range of Months_on_book: [13, 56]
range of Total_Relationship_Count: [1, 6]
range of Months_Inactive_12_mon: [0, 6]
range of Contacts_Count_12_mon: [0, 6]
range of Credit_Limit: [1438.3, 34516.0]
range of Total_Revolving_Bal: [0, 2517]
range of Avg_Open_To_Buy: [3.0, 34516.0]
range of Total_Amt_Chng_Q4_Q1: [0.0, 3.397]
range of Total_Trans_Amt: [510, 18484]
range of Total_Trans_Ct: [10, 139]
range of Total_Ct_Chng_Q4_Q1: [0.0, 3.714]
range of Avg_Utilization_Ratio: [0.0, 0.9990000000000001]


In [22]:
for col in numerical_cols:
    print(f'range of {col}: [{data[col].min()}, {data[col].max()}]')

range of CLIENTNUM: [708082083, 828343083]
range of Customer_Age: [26, 73]
range of Dependent_count: [0, 5]
range of Months_on_book: [13, 56]
range of Total_Relationship_Count: [1, 6]
range of Months_Inactive_12_mon: [0, 6]
range of Contacts_Count_12_mon: [0, 6]
range of Credit_Limit: [1438.3, 34516.0]
range of Total_Revolving_Bal: [0, 2517]
range of Avg_Open_To_Buy: [3.0, 34516.0]
range of Total_Amt_Chng_Q4_Q1: [0.0, 3.397]
range of Total_Trans_Amt: [510, 18484]
range of Total_Trans_Ct: [10, 139]
range of Total_Ct_Chng_Q4_Q1: [0.0, 3.714]
range of Avg_Utilization_Ratio: [0.0, 0.9990000000000001]


- CLIENTNUM is not important as it is supposed to be a categorical variable
- Customer age ranges from 26 to 73 years. There are no minor accounts
- No unusual dependent count values
- Clients who have been a customer of the bank for atleast 1 year have been considered going upto a maximum of ~4.5 years
- Maximum products used by customers are 6
- Customers with atmost 6 months of inactivity have been considered
- There is a contact between the bank and the customer atmost 6 times in the last 1 year
- Min credit limit is ~\\$1400 and maximum goes upto ~\\$34K
- Total revolving balance doesn't go higher than \\$2517
- But the difference between credit limit and the present balance has a maximum value as high as maximum credit limit
- Total value of transactions goes to a maximum value of 18K in the last 12 months and maximum number of transactions is 140
- There is almost 400\% change in transaction amount from Q1 to Q4 